staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -436,6 +437,10 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 					klog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedTypeName, err)
 				}
 			}
+			if apierrors.IsTooManyRequests(err) {
```

Comment:
would it make sense to put this in the switch statement and log differently?

Reply:
good point - done